packages/core/src/components/AutocompleteField/AutocompleteField.jsx

```diff
@@ -0,0 +1,175 @@
+      ...fieldProps
+    } = this.props;
+
+    return (
+      <Downshift
+        itemToString={itemToString}
+        onChange={onChange}
+        onStateChange={onStateChange}
+        render={({
+          clearSelection,
+          getInputProps,
+          getItemProps,
+          highlightedIndex,
+          isOpen,
+          inputValue
+        }) => (
+          <div className="ds-u-display--flex ds-u-flex-wrap--wrap ds-c-autocomplete">
+            <FormLabel
+              className="ds-u-margin-top--0"
+              labelClassName="ds-c-autocomplete__label"
+              component="label"
+              fieldId={this.id}
+              hint={labelHint}
+            >
+              {labelText}
+            </FormLabel>
+
+            <input
+              {...getInputProps({
+                className: 'ds-c-field ds-c-autocomplete__input',
+                disabled: disabled,
+                id: this.id,
+                ref: fieldRef,
+                ...fieldProps
+              })}
+            />
+
+            {isOpen ? (
+              <div className="ds-u-border--1 ds-u-padding--1 ds-c-autocomplete__list">
+                <ul className="ds-c-list--bare">
+                  {items
+                    .filter(
+                      item =>
+                        !inputValue ||
+                        item.name
+                          .toLowerCase()
+                          .includes(inputValue.toLowerCase())
+                    )
+                    .map((item, index) => (
+                      <li
+                        className={
+                          highlightedIndex === index
+                            ? 'ds-u-padding--1 ds-c-autocomplete__list-item ds-c-autocomplete__list-item--active'
+                            : 'ds-u-padding--1 ds-c-autocomplete__list-item'
+                        }
+                        key={item.id}
+                        {...getItemProps({ item })}
+                      >
+                        {itemToString(item)}
+                      </li>
+                    ))}
+                </ul>
+              </div>
+            ) : null}
+
+            <Button
+              aria-label={clearAriaLabel}
+              className="ds-u-margin-left--auto ds-u-padding-right--0 ds-c-autocomplete__button"
+              href="javascript:void(0);"
+              onClick={clearSelection}
+              size="small"
+              variation="transparent"
+            >
+              {clearInputText}
+            </Button>
+          </div>
+        )}
+      />
+    );
+  }
+}
+
+AutocompleteField.defaultProps = {
+  clearAriaLabel: 'Clear typeahead and search again',
+  clearInputText: 'Search again',
+  itemToString: noop,
+  labelHint:
+    'This is an autocomplete field. Begin typing to search for relevant information. The number of results will be updated as you type.',
+  labelText: 'Generic autocomplete label',
+  onChange: noop,
+  onStateChange: noop
```

Comment:
Do we need this to be `noop` or could these be `undefined` (no default) until the user defines them?

Reply:
I don't have strong feelings either way. I have used `noop` in the past so I can make function props like `onChange` required, test more easily. I'll remove them, and update the tests.

Reply:
I think if it's required in order to work, I'd argue it's better not to have a default so that a warning is logged to the console when a developer is implementing it. Also one less dependency if we don't need `noop` 😛

Reply:
That actually works better, setting to undefined. I'll use that for all patterns in the future.